test/unit.js

```diff
@@ -154,6 +154,19 @@ suite("unit", () => {
     invalidStmt(1, new Shift.FunctionDeclaration(ID, [new Shift.Identifier("if"), ID], new Shift.FunctionBody([], [])));
   });
 
+  test("IfStatement with null `alternate` must not be the `consequent` of an IfStatement with a non-null `alternate`", () => {
+    validStmt(new Shift.IfStatement(EXPR, new Shift.DoWhileStatement(new Shift.IfStatement(EXPR, STMT, null), EXPR), STMT));
+    validStmt(new Shift.IfStatement(EXPR, new Shift.IfStatement(EXPR, STMT, STMT), STMT));
```

Comment:
You're missing a valid test with a `null` alternate on the outer `IfStatement` and a consequent that would otherwise be problematic.